apps/src/lib/kits/maker/CircuitPlaygroundBoard.js

```diff
@@ -86,13 +103,51 @@ export default class CircuitPlaygroundBoard extends EventEmitter {
         this.serialPort_ = serialPort;
         this.fiveBoard_ = board;
         this.fiveBoard_.samplingInterval(100);
+        if (experiments.isEnabled('detect-board')) {
+          this.detectFirmwareVersion(playground);
+        }
         resolve();
       });
       board.on('error', reject);
       playground.on('error', reject);
     });
   }
 
+  /**
+   * Helper to detect the firmware (major and minor) version of the given playground
+   * @param playground
+   */
+  detectFirmwareVersion(playground) {
+    playground.queryFirmware(() => {
+      console.log(
+        playground.firmware.version.major +
+          '.' +
+          playground.firmware.version.minor
+      );
+    });
+  }
+
+  /**
+   * Detects the type of board plugged into the serial port
+   */
+  detectBoardType() {
+    return findPortWithViableDevice().then(port => {
```

Comment:
Do we need to re-list the devices here, or can we get it from the promise chain?